opm/grid/cpgrid/Iterators.hpp

```diff
@@ -52,9 +52,10 @@ namespace Dune
         /// This could have been a random access iterator, perhaps we will
         /// use a facade to do this later.
         template<int cd, PartitionIteratorType pitype>
-        class Iterator : public EntityPointer<cd>
+        class Iterator : public Entity<cd>
```

Comment:
A little dangerous, as it could let code such as this compile:
```
void foo(const Entity& e) { ... }
Iterator it;
...
foo(it);
```
where it is really `foo(*it)` that should be required for portability to other grids for example. This is not a very serious defect, and it is not a change since the old `EntityPointer` inherited publicly from `Entity` anyway, but worth considering.

Reply:
Seems like a flow in the original design. Should probably be changed, but not in this PR.